{
  "uid" : "613b060e460b4d68",
  "name" : "1.3.4 Check response has key job",
  "fullName" : "tests.create_users_test.TestCreateUser#test_post_create_user_key_name_is_in_response",
  "historyId" : "8c13af4cca75082904cd6ee642883bc5",
  "time" : {
    "start" : 1708427532754,
    "stop" : 1708427533007,
    "duration" : 253
  },
  "status" : "skipped",
  "statusMessage" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
  "statusTrace" : "self = <tests.create_users_test.TestCreateUser object at 0x7f2ea4d14e10>\nkey_name = 'job'\n\n    @pytest.mark.positive\n    @pytest.mark.parametrize('key_name', data.LIST_KEY)\n    @pytest.mark.xfail(reason='Bag')\n    def test_post_create_user_key_name_is_in_response(self, key_name):\n        allure.dynamic.title(f'1.3.{self.data.LIST_KEY.index(key_name) + 1} Check response has key {key_name}')\n        headers = self.data.CREATE_USER\n        response = BaseRequests.post(url='api/users', headers=headers)\n>       Assertions.assert_key_name_is_in_response(response, key_name)\n\ntests/create_users_test.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [201]>, key_name = 'job'\n\n    @staticmethod\n    def assert_key_name_is_in_response(response: Response, key_name):\n        \"\"\"Check if the required key is in the response\"\"\"\n        json_value = response.json()\n        actual_key = json_value.get(key_name)\n        with allure.step(f'The key {actual_key} exist'):\n>           assert actual_key is not None, f'The key {actual_key} does not exist'\nE           AssertionError: The key None does not exist\n\nbase/assertions.py:36: AssertionError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1708427531431,
      "stop" : 1708427531446,
      "duration" : 15
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "skipped",
    "statusMessage" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
    "statusTrace" : "self = <tests.create_users_test.TestCreateUser object at 0x7f2ea4d14e10>\nkey_name = 'job'\n\n    @pytest.mark.positive\n    @pytest.mark.parametrize('key_name', data.LIST_KEY)\n    @pytest.mark.xfail(reason='Bag')\n    def test_post_create_user_key_name_is_in_response(self, key_name):\n        allure.dynamic.title(f'1.3.{self.data.LIST_KEY.index(key_name) + 1} Check response has key {key_name}')\n        headers = self.data.CREATE_USER\n        response = BaseRequests.post(url='api/users', headers=headers)\n>       Assertions.assert_key_name_is_in_response(response, key_name)\n\ntests/create_users_test.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nresponse = <Response [201]>, key_name = 'job'\n\n    @staticmethod\n    def assert_key_name_is_in_response(response: Response, key_name):\n        \"\"\"Check if the required key is in the response\"\"\"\n        json_value = response.json()\n        actual_key = json_value.get(key_name)\n        with allure.step(f'The key {actual_key} exist'):\n>           assert actual_key is not None, f'The key {actual_key} does not exist'\nE           AssertionError: The key None does not exist\n\nbase/assertions.py:36: AssertionError",
    "steps" : [ {
      "name" : "The key None exist",
      "time" : {
        "start" : 1708427533006,
        "stop" : 1708427533006,
        "duration" : 0
      },
      "status" : "failed",
      "statusMessage" : "AssertionError: The key None does not exist\n",
      "statusTrace" : "  File \"/home/runner/work/reqres/reqres/base/assertions.py\", line 36, in assert_key_name_is_in_response\n    assert actual_key is not None, f'The key {actual_key} does not exist'\n           ^^^^^^^^^^^^^^^^^^^^^^\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Create User"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.xfail(reason='Bag')"
  }, {
    "name" : "tag",
    "value" : "positive"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "create_users_test"
  }, {
    "name" : "subSuite",
    "value" : "TestCreateUser"
  }, {
    "name" : "host",
    "value" : "fv-az775-764"
  }, {
    "name" : "thread",
    "value" : "2209-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.create_users_test"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "key_name",
    "value" : "'job'"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 0,
        "skipped" : 9,
        "passed" : 0,
        "unknown" : 0,
        "total" : 9
      },
      "items" : [ {
        "uid" : "36d7554b9880824",
        "reportUrl" : "https://Nat754.github.io/reqres/11//#testresult/36d7554b9880824",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708425059198,
          "stop" : 1708425059448,
          "duration" : 250
        }
      }, {
        "uid" : "f6d12d4ce35a12ce",
        "reportUrl" : "https://Nat754.github.io/reqres/10//#testresult/f6d12d4ce35a12ce",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420917001,
          "stop" : 1708420917239,
          "duration" : 238
        }
      }, {
        "uid" : "888be7b4c1053fce",
        "reportUrl" : "https://Nat754.github.io/reqres/9//#testresult/888be7b4c1053fce",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420826072,
          "stop" : 1708420826319,
          "duration" : 247
        }
      }, {
        "uid" : "dd45f1581b32ee15",
        "reportUrl" : "https://Nat754.github.io/reqres/8//#testresult/dd45f1581b32ee15",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708420746239,
          "stop" : 1708420746481,
          "duration" : 242
        }
      }, {
        "uid" : "2ebd3e34a676ec15",
        "reportUrl" : "https://Nat754.github.io/reqres/7//#testresult/2ebd3e34a676ec15",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708418351489,
          "stop" : 1708418351855,
          "duration" : 366
        }
      }, {
        "uid" : "522ca0772fda2a27",
        "reportUrl" : "https://Nat754.github.io/reqres/6//#testresult/522ca0772fda2a27",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708412184728,
          "stop" : 1708412184970,
          "duration" : 242
        }
      }, {
        "uid" : "d0baac8669a3bd27",
        "reportUrl" : "https://Nat754.github.io/reqres/5//#testresult/d0baac8669a3bd27",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708388621937,
          "stop" : 1708388622167,
          "duration" : 230
        }
      }, {
        "uid" : "a8cfd4092e2f85d8",
        "reportUrl" : "https://Nat754.github.io/reqres/4//#testresult/a8cfd4092e2f85d8",
        "status" : "skipped",
        "statusDetails" : "XFAIL Bag\n\nAssertionError: The key None does not exist",
        "time" : {
          "start" : 1708385916653,
          "stop" : 1708385916997,
          "duration" : 344
        }
      } ]
    },
    "tags" : [ "@pytest.mark.xfail(reason='Bag')", "positive" ]
  },
  "source" : "613b060e460b4d68.json",
  "parameterValues" : [ "'job'" ]
}